src/interface/sparse_pack.hpp

```diff
@@ -98,6 +99,17 @@ struct base_t {
   KOKKOS_INLINE_FUNCTION
   explicit base_t(int idx1) : idx(idx1) {}
 
+  template <typename... Args, REQUIRES(all_implement<integral(Args...)>::value)>
+  /*
+    for 2D:, (M, N),
+    idx(m, n) = N*m + n
+    for 3D: (L, M, N)
+    idx(l, m, n) = (M*l + m)*N + n
+                 = l*M*N + m*N + n
+   */
```

Comment:
I think we should keep this comment but it should probably be above the `template<...` line

Reply:
agreed

Reply:
done